graphql_compiler/compiler/compiler_frontend.py

```diff
@@ -1067,20 +1057,53 @@ def graphql_to_ir(schema, graphql_string, type_equivalence_hints=None):
 
     In the case of implementation bugs, could also raise ValueError, TypeError, or AssertionError.
     """
-    graphql_string = _preprocess_graphql_string(graphql_string)
-    try:
-        ast = parse(graphql_string)
-    except GraphQLSyntaxError as e:
-        raise GraphQLParsingError(e)
-
-    validation_errors = _validate_schema_and_ast(schema, ast)
-
+    validation_errors = validate_schema_and_ast(schema, ast)
     if validation_errors:
         raise GraphQLValidationError(u'String does not validate: {}'.format(validation_errors))
 
-    if len(ast.definitions) != 1:
-        raise AssertionError(u'Unsupported graphql string with multiple definitions, should have '
-                             u'been caught in validation: \n{}\n{}'.format(graphql_string, ast))
-    base_ast = ast.definitions[0]
-
+    base_ast = get_only_query_definition(ast, GraphQLValidationError)
     return _compile_root_ast_to_ir(schema, base_ast, type_equivalence_hints=type_equivalence_hints)
+
+
+def graphql_to_ir(schema, graphql_string, type_equivalence_hints=None):
+    """Convert the given GraphQL string into compiler IR, using the given schema object.
+
+    Args:
+        schema: GraphQL schema object, created using the GraphQL library
+        graphql_string: string containing the GraphQL to compile to compiler IR
```

Comment:
For now, it has to be a `query`. But in principle, in the future it could be a `subscription` or `mutation`, and should be compiled appropriately. I'd leave it as `graphql_string`.